client/tests/integration_test.rs

```diff
@@ -0,0 +1,160 @@
+use std::convert::Infallible;
+
+use futures::{channel::oneshot, future};
+use hyper::{Body, Response, Server};
+use serde::Deserialize;
+use tokio::task::JoinHandle;
+use warp::{Filter, Rejection};
+use warp_json_rpc::Builder;
+
+use casper_client::{DeployStrParams, PaymentStrParams, SessionStrParams};
+use casper_node::rpcs::{
+    account::{PutDeploy, PutDeployParams},
+    chain::{GetStateRootHash, GetStateRootHashParams},
+    info::{GetDeploy, GetDeployParams},
+    RpcWithOptionalParams, RpcWithParams,
+};
+
+fn test_filter<P>(
+    method: &'static str,
+) -> impl Filter<Extract = (Response<Body>,), Error = Rejection> + Copy
+where
+    for<'de> P: Deserialize<'de> + Send,
+{
+    warp_json_rpc::filters::json_rpc()
+        .and(warp_json_rpc::filters::method(method))
+        .and(warp_json_rpc::filters::params::<P>())
+        .map(|builder: Builder, _params: P| builder.success(()).unwrap())
+}
+
+struct MockServerHandle {
+    graceful_shutdown: oneshot::Sender<()>,
+    server_joiner: JoinHandle<Result<(), hyper::error::Error>>,
+}
+
+impl MockServerHandle {
+    async fn kill(self) {
+        let _ = self.graceful_shutdown.send(());
+        let _ = self.server_joiner.await;
+    }
+}
```

Comment:
Let's make this RAII, so we don't have to worry about calling `kill()` regardless of where the test exits (in case it exists early e.g.):

```suggestion
struct MockServerHandle {
    graceful_shutdown: Option<oneshot::Sender<()>>,
    server_joiner: Option<JoinHandle<Result<(), hyper::error::Error>>>,
}

impl Drop for MockServerHandle {
    fn drop(&mut self) {
        let _ = self.graceful_shutdown.take().unwrap().send(());
        let joiner = self.server_joiner.take().unwrap();
        futures::executor::block_on(async {
            let _ = joiner.await;
        });
    }
}
```

Reply:
Thanks for that.